DRODLib/DbRooms.cpp

```diff
@@ -3592,7 +3594,9 @@ CMonster* CDbRoom::GetMonsterOfType(const UINT wType) const
 			CCharacter *pCharacter = DYN_CAST(CCharacter*, CMonster*, pMonster);
 			if (pCharacter->IsVisible()) //Character must be in room to count.
 			{
-				if (wType == M_CHARACTER || pCharacter->GetIdentity() == wType)
+				if (wType == M_CHARACTER)
+					return pMonster;
+				else if (pCharacter->GetIdentity() == wType && !bIgnoreCharacterAppearance)
```

Comment:
Technically don't need the `else` here.  I think it's easier to read without it, but YMMV.

Reply:
No disagreement here but since you merged it it's a little bit too late to change :D.

Reply:
Yeah, no worries.  But!  It's never too late to change. Not suggesting you need to put up another PR for that change though.